### PR TITLE
Adding tooltips for sort and filter fields

### PR DESCRIPTION
**Pull request by erundle**

### CODE REVIEW

File: dist/angular-patternfly.js [hunk 1]
```diff
@@ -3205,6 +3205,26 @@ angular.module('patternfly.sort').directive('pfSimpleSort', function () {
...
+          }
+        } else {
+          if ($scope.config.isAscending) {
+            tooltip = 'Sort alphabetically, from A to Z';
+          } else {
+            tooltip = 'Sort alphabetically, from Z to A';
+          }
+        }
+
+        $scope.sortTooltip = tooltip;
```
Comment (waldenraines) [inline dist/angular-patternfly.js]: What about putting these in the HTML?  Seems odd to have this sort of stuff in a controller.

File: src/sort/simple-sort-directive.js [hunk 2]
```diff
@@ -154,6 +154,26 @@ angular.module('patternfly.sort').directive('pfSimpleSort', function () {
...
+          }
+        } else {
+          if ($scope.config.isAscending) {
+            tooltip = 'Sort alphabetically, from A to Z';
+          } else {
+            tooltip = 'Sort alphabetically, from Z to A';
+          }
+        }
+
+        $scope.sortTooltip = tooltip;
```
Comment (waldenraines) [inline src/sort/simple-sort-directive.js]: What about putting these in the HTML? Seems odd to have this sort of stuff in a controller.

Comment (erundle) [inline src/sort/simple-sort-directive.js]: I don't think we really want that many ng-ifs sprinkled in the HTML unless you have another idea. We have the icon logic in the controller so for consistency reasons I would suggest they stay together.

**Comment by dtaylor113** [comment]
LGTM

File: src/sort/simple-sort-directive.js [hunk 3]
```diff
@@ -154,6 +154,26 @@ angular.module('patternfly.sort').directive('pfSimpleSort', function () {
     },
     templateUrl: 'sort/simple-sort.html',
     controller: function ($scope) {
+      var setSortTooltip = function () {
```
Comment (jeff-phillips-18) [inline src/sort/simple-sort-directive.js]: This name is a bit confusing, more specifically this is the sort direction tooltip.

Comment (erundle) [inline src/sort/simple-sort-directive.js]: Changed to setSortDirectionTooltip and squashed

**Comment by jeff-phillips-18** [comment]
LGTM

**Comment by erundle** [comment]
@waldenraines can this be merged?

**Comment by waldenraines** [comment]
@erundle I personally would like to see the tooltips in the HTML itself rather than the controller, what is the opposition to ng-ifs in the HTML?  I won't hold this PR up on this point but to me it's cleaner.

**Comment by erundle** [comment]
I would prefer not to repeat this markup 4 times:

<button class="btn btn-link" type="button"  ng-click="changeDirection()" tooltip="{{ sortDirectionTooltip }}" tooltip-placement="bottom"><span class="sort-direction" ng-class="getSortIconClass()"></span></button>

Granted with having the ng-ifs in the HTML we can also get rid of the getSortIconClass method I just find this harder to read and more error prone but that is just me.

File: src/sort/simple-sort-directive.js [hunk 4]
```diff
@@ -154,6 +154,26 @@ angular.module('patternfly.sort').directive('pfSimpleSort', function () {
     },
     templateUrl: 'sort/simple-sort.html',
     controller: function ($scope) {
+      var setSortDirectionTooltip = function () {
+        var tooltip = '';
+
+        if ($scope.config.currentField.sortType === 'numeric') {
+          if ($scope.config.isAscending) {
+            tooltip = 'Sort numerically, from least to greatest';
```
Comment (waldenraines) [inline src/sort/simple-sort-directive.js]: When I hover over a button, I expect the tooltip to tell me what pressing the button would do, not what the current state of the button is.  Here we are doing the latter and I just want to make sure that is intentional.

Comment (jeff-phillips-18) [inline src/sort/simple-sort-directive.js]: I have to agree with @waldenraines here. I'm not really sure why we are setting the tooltip to explain what the current state of the button means. Shouldn't the tooltip just be "Change the current sort direction" always?

**Comment by waldenraines** [comment]
> I would prefer not to repeat this markup 4 times:

Me too.  If I were to design this I may have done so by wrapping the tooltip directive in a "directional tooltip" directive that handles (and hides away) this behavior.  I suppose it just seems odd to me to have logic for a tooltip inside of the sort controller but, like I said, I won't hold this PR up because of that.

Do you think we'll have to copy this logic elsewhere in the future?  If so then this approach would make sense.

File: src/sort/simple-sort-directive.js (hunk 4 above)
Comment (erundle) [inline src/sort/simple-sort-directive.js]: I asked @serenamarie125 to comment since this text came from her

**Comment by erundle** [comment]
I don't know if we will have more of this in the future but creating a directive for this seems like overkill for a handful of if blocks. If we were to take that approach we would want to do that same thing for the sort icon itself which means this simple-sort-directive.js is pretty much paired down to nothing. I could see taking that approach if the file size was large and these components were complex but right now that isn't the case. IMO having everything in one file makes it easier for the developer to understand what is going on with a code base of this size.

File: src/sort/simple-sort-directive.js (hunk 4 above)
Comment (serenamarie125) [inline src/sort/simple-sort-directive.js]: Please pull the tooltips from the sort icons themselves, and only have tips on Sort by and Filter by.

**Comment by erundle** [comment]
I talked with @serenamarie125 and we are going to pull the tooltip from the sort icon all together. I made the change and squashed the commit so this should be G2G.

**Pull request merged**